go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+package databricks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// ColumnBuilder is an interface for building Arrow arrays
+type ColumnBuilder interface {
+	Append(value interface{}) error
```

Comment:
Let's change the `Append` interface a bit:

1) return bool instead of error
2) define value != nil as a pre-condition

The loop that calls Append can check if value is null and call `col_builder.Builder().AppendNull()` and only call `col_builder.Append()` otherwise.

Then if it gets `false` as return value, you can produce an error that mentions that type that was expected (you can query the `col_builder.Builder()` for that) is not what you got from value.

```suggestion
	// \pre: value != nil
	Append(value interface{}) bool
```

That means we have to write the null checking `if` and build the error object only once in the source code.